chapters/en/slides/module1/module1_02.Rmd

```diff
@@ -0,0 +1,280 @@
+For example,
+you would say "color my data by the column 'country'"
+instead of "go through this data frame and plot any observations of country1 in blue, any observations of country in red, etc".
+
+Declarative visualization tools lets you think about **data and relationship**,
+rather than **plot construction details**.
+A key idea is that you declare links between data fields and visual encoding channels,
+such as the x-axis, y-axis, color, *etc*.
+The rest of the plot details are handled automatically.
+
+---
+
+## The Python plotting landscape
+
+<img src="/module1/py-plotting-landscape.png" alt="The Python plotting landscape" width="100%"></img>
+
+Notes:
+In this image you can see the most commonly used Python plotting packages.
+There are many more,
+but these are the ones you are the most likely to hear about,
+so it is good to know that they exist.
+As you can see there are several high and low level language,
+so how to we chose?
+In this course we have chosen to use Altair,
+because it is a powerful high level visualization tool 
+with a clear and consistent syntax
+that also allows us to add interactive components to our plots,
+such as tooltips and selections.
+
+---
+
+## A high level grammar of graphics helps us compose plots effectively
+
+- Simple grammatical components combine to create visualizations.
+- The Altair visualization grammar consist of three main components:
+    1. Create a chart.
+    2. Add a graphical mark.
+    3. Encode dataframe columns as visual channels.
+- In code, it looks like this: `Chart(data).mark().encode(x, y, ...)`.
+
+Notes:
+The declarative plotting concept can be implemented in different ways.
+Here we will describe how a visualization grammar
+can be used similarly to how grammar in regular language
+is used to build complex linguistic constructs.
+A wide range of simple to sophisticated visualizations can be created using a concise grammar.
+Thanks to this functional way of interfacing with data,
+only minimal changes are required if the underlying data change or to change the type of plot.
+
+One of the most prominent declarative statistical visualization libraries is [Altair](https://altair-viz.github.io/).
+Altair offer a powerful and concise visualization grammar for quickly building a wide range of statistical graphics.
+In brief, you first create a chart,
+then you indicate which graphical marks should represent the data (points, lines, etc)
+and finally you encode your dataframe colums as different visual channels (x, y, color, etc).
+
+Enough talking,
+let's code!
+
+---
+
+## Sample data can be found in Altair's companion package vega_datasets
+
+```{python}
+from vega_datasets import data
+
+cars = data.cars()
+cars
+```
+
+Notes:
+When specifying data in Altair,
+it is important that dataframes are in the ["tidy"](http://vita.had.co.nz/papers/tidy-data.html) format.
+They should consist of a set of named data *columns* (also called *fields* or *variables*)
+with one feature each
```

Comment:
Here we have just used 3 names in one sentence to describe a columnar variable. There is already a lot of jargon in this course, let's stick with one name for this. I would vote for variable.

Reply:
I have clarified this to:

> ...
> and a set of named data *columns*
> with one feature each
> (you might also have heard these called *fields* or *variables*,
> but we will stick to *columns* for clarity).

I stuck to "columns" because I think it is clearer since we are referring to the other axis as "rows", but we can discuss more. But in general I think we should use the same as in the prerequisite MCL course, @hfboyce which nomenclature did you use there?

Reply:
Writing the faceting and EDA chapters, it would be nice to use "variable" since there now already is something called "column" in the plot grid. @hfboyce Can I change this? Or what did you use previously?

Reply:
In programing in python they should be comfortable with variable or column. We used both interchangeably in that course. In the ML we used "feature" as well.